tensorflow/python/data/ops/dataset_ops.py

```diff
@@ -1165,7 +1165,20 @@ def i(...):
     some but not all Python code.
 
     2) Use `tf.py_function`, which allows you to write arbitrary Python code but
-    will generally result in worse performance than 1).
+    will generally result in worse performance than 1). For example:
+    
+    ```python
+    d = tf.data.Dataset.from_tensor_slices(['hello', 'world'])
+    
+    # transform a string tensor to upper case string using a Python function
+    def upper_case_fn(t: tf.Tensor) -> str:
+        return t.numpy().decode('utf-8').upper()
+    
+    result = d.map(lambda x: tf.py_function(func=upper_case_fn,
```

Comment:
I would prefer the following for consistency for the rest of the documentation:

```
d.map(lambda x: tf.py_function(...))  # ==> [ "HELLO", "WORLD" ]
```

Reply:
While I completely agree with the consistency point, the left-out parameters of py_function hide the exact way how to call the function which might not be obvious to an unexperienced person. The details must be then searched in the documentation of py_function but that one shows a custom logarithm function executed in a TF1-style session which is rather confusing. I believe the crisp explicit example might be helpful (yet the output might be shortened, that's fine).

How about the following?

```python
d = tf.data.Dataset.from_tensor_slices(['hello', 'world'])
    
# transform a string tensor to upper case string using a Python function
def upper_case_fn(t: tf.Tensor) -> str:
    return t.numpy().decode('utf-8').upper()
    
d.map(lambda x: tf.py_function(func=upper_case_fn,
      inp=[x], Tout=tf.string))  # ==> [ "HELLO", "WORLD" ]
```

Reply:
I was not suggesting to left out the parameters. I was just lazy copying them.

Your suggestion looks good to me.

Reply:
Fair enough, thanks!